File: public/static/templates/nodejs/event/AbstractEvent.js
const streamhandler = require('../streamhandler/producer')
class AbstractEvent{
    constructor(){
        let date = new Date;
        this.eventType = this.constructor.name;
        this.timeStamp = date;
    }
    ToJson(){
        return JSON.stringify(this);
    }
    async Publish(){
        var msg = this.ToJson();
        await streamhandler.producer.send({
            topic: 'mall',
            messages: [
              {value: msg}
            ],
        })
    }
}

module.exports = AbstractEvent;

File: public/static/templates/python/AbstractEvent.py
from datetime import datetime

import KafkaProcessor

class AbstractEvent:
	eventType : str
	timeStamp : str

	def __init__(self):
		self.eventType = self.__class__.__name__
		self.timeStamp = str(datetime.now())

	def ToJson(self):
		return self.__dict__

	def Publish(self):
		msg = self.ToJson()

		processor = KafkaProcessor.streamhandler
		processor.produce(msg)